R/loadFromMetaphlan.R

```diff
@@ -150,18 +150,20 @@ loadFromMetaphlan <- function(file, sample_meta = NULL, phy_tree = NULL, ...){
 
 # Check that metaphlan file contains correct information
 .check_metaphlan <- function(data){
+    # Getting column indices for rowdata and assay
+    col_idx <- unlist(lapply(data, is.character))
     # Get rowdata columns
-    rowdata_columns <- data[ , 1:2]
+    rowdata_columns <- data[ , col_idx, drop=FALSE]
     # Get columns that go to assay
-    assay_columns <- data[ , 3:ncol(data)]
+    assay_columns <- data[ , !col_idx]
     # Initialize result 
     result <- TRUE
     
     # Check rowdata column names that they contain right information, and check that 
     # rest of the columns represents abundances in samples.
     # If these requirements are met, give FALSE. Otherwise, give TRUE.
     if( any(colnames(rowdata_columns) %in% "clade_name") && 
-        any(grepl("id", colnames(rowdata_columns))) && 
+        #any(grepl("id", colnames(rowdata_columns))) && 
```

Comment:
If this check is not needed¨, it's better to remove than comment

 "does metaphlan file has a column that includes "id"..."

Reply:
With the demo test done earlier and the discussion at #306  it seems for now the metaphlan4 is not producing that column name.